Orange/widgets/data/owselectbydataindex.py

```diff
@@ -62,6 +69,18 @@ def set_data_subset(self, data):
         self.infoBoxExtraData.setText(self.data_info_text(data))
 
     def handleNewSignals(self):
+        if self.data and self.data_subset:
+            data_list = [('Data', self.data), ('Data subset', self.data_subset)]
+            summary = f"{len(self.data)}, {len(self.data_subset)}"
+            details = format_multiple_summaries(data_list)
+            self.info.set_input_summary(summary, details, format=Qt.RichText)
+        elif not self.data and not self.data_subset:
+            self.info.set_input_summary(self.info.NoInput)
+        else:
```

Comment:
The two inputs are considered equivalent, so I would omit this `else` and replace the upper condition `self.data and self.data_subset` with `self.data or self.data_subset`.

In case when only one dataset is on the input, just show 0 in the summary.
Something like:
```
        summary, detail, kwargs = self.info.NoInput, "", {}
        if self.data or self.data_subset:
            n_data = len(self.data) if self.data else 0
            n_data_subset = len(self.data_subset) if self.data_subset else 0
            summary = f"{self.info.format_number(n_data)}, " \
                      f"{self.info.format_number(n_data_subset)}"
            kwargs = {"format": Qt.RichText}
            detail = format_multiple_summaries([
                ("Data", self.data),
                ("Data subset", self.data_subset)
            ])
        self.info.set_input_summary(summary, detail, **kwargs)
```